Add research notice to dev docs

yochannah opened the pull request:
This replaced #568 - apologies for making a messy driveby commit. The date reflects the month the notice made it into the docs, not the date the study started, so the date should be correct. Thanks for your patience and help :) 
----
@skitt is aware of this study :) Please let me know if there's a better place to put this notice, I thought the dev docs might be the easiest place for people to spot it.

Signed-off-by: Yo Yehudi <yochannah.yehudi@postgrad.manchester.ac.uk>

Commit:
Add research notice to dev docs

@skitt is aware of this study :) Please let me know if there's a better place to put this notice, I thought the dev docs might be the easiest place for people to spot it.

Signed-off-by: Yo Yehudi <yochannah.yehudi@postgrad.manchester.ac.uk>

Commit:
correct date in research notice

Signed-off-by: Yo Yehudi <yochannah.yehudi@postgrad.manchester.ac.uk>

dfarrell07 commented on src/content/development/_index.en.md:
```diff
@@ -18,3 +18,14 @@ pre = "<b>4. </b>"
   * [Adding Shipyard to a Project](shipyard/first-time)
   * [Advanced Features](shipyard/advanced)
+
+### Research notice
```
Markdownlint is failing because you need a blank line after the section header.

> src/content/development/_index.en.md: 22: MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Research notice"]

skitt replied:
```suggestion
### Research notice

```

skitt commented on src/content/development/_index.en.md:
```diff
@@ -18,3 +18,14 @@ pre = "<b>4. </b>"
+### Research notice
+Please note that this repository is participating in a study into sustainability
+ of open source projects. Data will be gathered about this project's github repositories for
+ approximately the next 12 months, starting from September 2020.
```
If this is supposed to be the date at which the notice is added to the docs, it should be

```suggestion
 approximately the next 12 months, starting from September 2021.
```

dfarrell07 commented on src/content/development/_index.en.md:
```diff
@@ -18,3 +18,14 @@ pre = "<b>4. </b>"
+
+### Research notice
+Please note that this repository is participating in a study into sustainability
+ of open source projects. Data will be gathered about this project's github repositories for
```
```suggestion
 of open source projects. Data will be gathered about this project's GitHub repositories for
```

While you're making the required fix for CI, one little nit here. :)

dfarrell07 committed:
Set yamllint strict flag correctly

Setting yamllint to exit non-zero on warnings doesn't work from the
.yamllint config file.

Set the --strict flag at CLI invocation, which in this case is handled
by the GHA and must be configured there.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>
Signed-off-by: Yo Yehudi <yochannah.yehudi@postgrad.manchester.ac.uk>

dfarrell07 committed:
Fix yamllint warnings

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>
Signed-off-by: Yo Yehudi <yochannah.yehudi@postgrad.manchester.ac.uk>

yochannah committed:
Update src/content/development/_index.en.md

Co-authored-by: Stephen Kitt <steve@sk2.org>
Signed-off-by: Yo Yehudi <yochannah.yehudi@postgrad.manchester.ac.uk>

yochannah committed:
Update src/content/development/_index.en.md

Co-authored-by: Daniel Farrell <dfarrell07@gmail.com>
Signed-off-by: Yo Yehudi <yochannah.yehudi@postgrad.manchester.ac.uk>

yochannah committed:
Update src/content/development/_index.en.md

Co-authored-by: Stephen Kitt <steve@sk2.org>
Signed-off-by: Yo Yehudi <yochannah.yehudi@postgrad.manchester.ac.uk>

yochannah commented:
Hopefully this should be fixed now - just waiting for the CI to do its stuff.

skitt committed:
Merge branch 'devel' into research-notice
